homeworks/03-data-structure/src/BinarySearchTree.ts

```diff
@@ -0,0 +1,26 @@
+import {BinaryTree, TreeNode, TraverseType} from "./BinaryTreeInterfaces"
+import {BinaryTreeClass} from "./BinaryTree";
+
+export class BinarySearchTree extends BinaryTreeClass<number> implements BinarySearchTree {

```

Comment:
No `implements BinarySearchTree`! You don't have a separate interface to implement. You reused the class interface. It's completely incorrect and doesn't make sense.